CONTRIBUTING.md

```diff
@@ -1,2 +1,132 @@
+feature/my-topic-branch or bug/fix-topic-branch
+```
+
+### Commit Message Conventions:
+
+Capitalize the subject line and do not end the subject with a period.
+Use the imperative mood in the subject line. `This commit will <your commit message>`.
+
+### Pull Request:
+
+- Bug branch pull request description will reference the issue# fixed.
+- Feature branch pull request description will explain what the feature is and changes proposed.
+
+## Sending a Pull Request
+
+PX Blue is a community with many projects, so pull requests are always welcome. If you are working on a complete change, please open an issue first to discuss it with the maintainers of PX Blue. Keep pull requests small to have the best chance of getting accepted.
+
+```
+Don't bundle more than one feature or bug fix per pull request.
+```
+
+## Get started
+
+1. Fork the PX Blue repository to your GitHub account.
+
+2. Clone the fork repository to your local machine and add upstream remote:
+
+```
+git clone https://github.com/<your-user-id>/<repository> -b <branch-name>
+cd <repository>
+git remote add upstream https://github.com/pxblue/<repository-name>
+```
+
+3. Synchronize your local branch with the upstream one:
+
+```
+git pull upstream <branch>
+```
+
+4. Install the dependencies:
+
+```
+yarn install
+```
+
+5. Create a new branch:
+
+```
+git checkout -b feature/my-topic-branch
+Or
+git checkout -b bug/my-topic-branch
+```
+
+6. Make changes, commit and push to your fork:
+
+```
+git push –set upstream origin <your branch name>
+```
+
+7. Go to [the repository](https://github.com/pxblue) and make a Pull Request.
+
+The maintainers of PX Blue will be notified and will review your pull request and either merge it, request changes to it or close it with an explanation.
+
+### How to increase the chance of Pull Requests being accepted
+
+Because we will only merge Pull Request for which all tests pass. The following items also need to be checked.
+
+- If TypeScript declarations were changed, `yarn typescript`
+- The code is formatted `yarn prettier`
+- The code is linted `yarn lint`
+- The code builds `yarn build`
+- The code passes tests `yarn test`
+- The Branch, Commit Message and Pull Request follows our pattern.
+
+### Verify the following:
+
+- The branch is targeted at `master` for ongoing development. We do our best to keep `master` clean with all tests passing. Code that lands in `master` must be compatible with the latest stable code. It may contain additional features, but no breaking changes. We should be able to release a new minor version from the tip of `master` at any time.
+- If a feature is being added:
+
+  - Explain why the feature needs to be added.
+
+- When adding new features or modifying existing, please include tests to confirm the new behavior.
+- If props were added or prop types were changed, the TypeScript declarations are updated.
+
+## Contribution Prerequisites
+
+- Node v10.17.0+
+- Yarn v1.17.0+
+- Latest PX Blue NPM packages installed.
```

Comment:
Is this necessary here? Presumably any repository that uses PX Blue packages already has them in the yarn lock.
Maybe we just say if you use any packages, make sure they are the latest version